### PR TITLE
[update] Work on e2e refactoring to Java - integrations-steps

### PR DESCRIPTION
**Pull request by Stefan365**
waiting for
integrations/detail

### CODE REVIEW

File: ui-tests/src/test/java/io/syndesis/qe/pages/integrations/edit/IntegrationEditPage.java [hunk 1]
```diff
@@ -8,18 +8,44 @@
...
+		switch (alias) {
+			case "Choose a Finish Connection": {
+				elementName = "h1";
+				//move this method to editPage object:
+				log.info("there must be edit editPage element");
+
+				break;
+			}
+			case "Add to Integration": {
+				elementName = "h1";
```
Comment (dsimansk) [inline ui-tests/src/test/java/io/syndesis/qe/pages/integrations/edit/IntegrationEditPage.java]: There's same value for both cases.

File: ui-tests/src/test/java/io/syndesis/qe/steps/integrations/IntegrationSteps.java [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+		}
+		detailPage2.getActionButton("Done").shouldBe(visible).click();
+	}
+
+	@Then("^she go trough whole list of integrations and check on detail if status match and appropriate actions are available$")
+	public void goTrouhListAndCheckDetails() {
+		ElementsCollection integrations = listComponent.getAllIntegrations();
+
+		for (SelenideElement integration : integrations) {
+//			const integration = listComponent.getAllIntegrations().get(i);
```
Comment (dsimansk) [inline ui-tests/src/test/java/io/syndesis/qe/steps/integrations/IntegrationSteps.java]: There're code snippets left-overs through the whole file. Do you want to keep them for reference?

File: ui-tests/src/test/java/io/syndesis/qe/common/World.java [hunk 3]
```diff
@@ -5,7 +5,6 @@
 /**
  * Created by sveres on 11/10/17.
  */
-//	TBD: by mastepane
 public class World {
```
Comment (dsimansk) [inline ui-tests/src/test/java/io/syndesis/qe/common/World.java]: Let's completely ditch the World class as not needed anymore.

File: ui-tests/src/test/java/io/syndesis/qe/pages/integrations/edit/IntegrationEditPage.java [hunk 4]
```diff
@@ -8,18 +8,37 @@
...
 
+@Slf4j
 public class IntegrationEditPage extends SyndesisPageObject {
-	
+
 	private static final class Element {
 		public static final By ROOT = By.cssSelector("syndesis-integrations-edit-page");
 	}
 
+	public void checkPageIsPresent(String alias) {
```
Comment (lvydra) [inline ui-tests/src/test/java/io/syndesis/qe/pages/integrations/edit/IntegrationEditPage.java]: @dsimansk,  @Stefan365  
Martin made (#181) class for validating pages based on SyndesiPageObject validate() method. I think its nice way how to standardize page presence validation. WDYT?

File: ui-tests/src/test/java/io/syndesis/qe/steps/integrations/IntegrationSteps.java [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+		IntegrationDetailPage detailPage2 = integrationDetailPageFactory.getDetailPage(status);
+
+		for (String action : detailPage2.actionsSet) {
+			log.info("Action: {}", action);
+			SelenideElement actionButton = detailPage2.getActionButton(action);
+			log.info("There should by button for {} action on {} status", action, status);
+			actionButton.shouldBe(visible);
+
+			log.info("Status on detail editPage should be equal to expected status");
+			assert status.equals(detailPage2.getStatus()) : "Status on detail editPage should be equal to expected status";
```
Comment (lvydra) [inline ui-tests/src/test/java/io/syndesis/qe/steps/integrations/IntegrationSteps.java]: This should probably by outside for.

File: ui-tests/src/test/java/io/syndesis/qe/pages/integrations/list/IntegrationByStatus.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package io.syndesis.qe.pages.integrations.list;
+
+/**
+ * Created by sveres on 11/16/17.
+ */
+public class IntegrationByStatus {
```
Comment (lvydra) [inline ui-tests/src/test/java/io/syndesis/qe/pages/integrations/list/IntegrationByStatus.java]: I'm not sure if there is purpose for this class. Is there?

File: ui-tests/src/test/java/io/syndesis/qe/steps/integrations/IntegrationSteps.java (hunk 5 above)
Comment (dsimansk) [reply ui-tests/src/test/java/io/syndesis/qe/steps/integrations/IntegrationSteps.java]: And let's use `AssertionJ.asserThat` for it also.

File: ui-tests/src/test/java/io/syndesis/qe/steps/integrations/IntegrationSteps.java (hunk 2 above)
Comment (Stefan365) [reply ui-tests/src/test/java/io/syndesis/qe/steps/integrations/IntegrationSteps.java]: they are to be deleted, before merge.

File: ui-tests/src/test/java/io/syndesis/qe/pages/integrations/list/IntegrationByStatus.java (hunk 6 above)
Comment (Stefan365) [reply ui-tests/src/test/java/io/syndesis/qe/pages/integrations/list/IntegrationByStatus.java]: sorry, this one is here by mistake

File: ui-tests/src/test/java/io/syndesis/qe/steps/integrations/IntegrationSteps.java [hunk 7]
```diff
@@ -0,0 +1,366 @@
...
+	}
+
+	@Then("^she clicks on integration in \"(\\w+)\" status and check on detail if status match and appropriate actions are available$")
+	public void clickOnIntegrationInStatus(String status) {
+		SelenideElement integrationByStatus = listComponent.getIntegrationByStatus(status);
+		integrationByStatus.shouldBe(visible).click();
+
+		//WAITING FOR IntegrationDetailPageFactory()
+		const integrationDetailPageFactory = new IntegrationDetailPageFactory();
+		IntegrationDetailPage detailPage2 = integrationDetailPageFactory.getDetailPage(status);
```
[the remark this answers is not in the document]
Comment (Stefan365) [reply ui-tests/src/test/java/io/syndesis/qe/steps/integrations/IntegrationSteps.java]: this part hasn't been refactored, I was waiting for integrations/detail pages part

File: ui-tests/src/test/java/io/syndesis/qe/steps/integrations/IntegrationSteps.java (hunk 5 above)
Comment (Stefan365) [reply ui-tests/src/test/java/io/syndesis/qe/steps/integrations/IntegrationSteps.java]: ok. to be done

File: ui-tests/src/test/java/io/syndesis/qe/pages/integrations/edit/IntegrationEditPage.java [hunk 8]
```diff
@@ -1,25 +1,55 @@
...
+@Slf4j
 public class IntegrationEditPage extends SyndesisPageObject {
-	
+
 	private static final class Element {
 		public static final By ROOT = By.cssSelector("syndesis-integrations-edit-page");
 	}
 
+	private static final class CssClasses {
+		public static final String TRASH_CLASS = "fa-trash";
```
Comment (lvydra) [inline ui-tests/src/test/java/io/syndesis/qe/pages/integrations/edit/IntegrationEditPage.java]: I'm not sure, but I think we should keep convention of storing only locators.

**Pull request merged**